app/src/main/java/io/appium/settings/Settings.java

```diff
@@ -79,26 +62,17 @@ public void onCreate(Bundle savedInstanceState) {
         setContentView(R.layout.main);
         Log.d(TAG, "Entering the app");
 
-        registerSettingsReceivers(Arrays.asList(
-                WiFiConnectionSettingReceiver.class,
-                AnimationSettingReceiver.class,
-                DataConnectionSettingReceiver.class,
-                LocaleSettingReceiver.class,
-                LocalesReader.class,
-                LocationInfoReceiver.class,
-                ClipboardReceiver.class,
-                BluetoothConnectionSettingReceiver.class,
-                UnpairBluetoothDevicesReceiver.class,
-                NotificationsReceiver.class,
-                SmsReader.class,
-                MediaScannerReceiver.class
-        ));
-
-        // https://developer.android.com/about/versions/oreo/background-location-limits
-        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
-            startForegroundService(ForegroundService.getForegroundServiceIntent(Settings.this));
+        String recordingAction = getIntent().getAction();
+        if (recordingAction != null && recordingAction.startsWith(ACTION_RECORDING_BASE)) {
+            Log.d(TAG, "Skip starting foreground service");
         } else {
-            LocationTracker.getInstance().start(this);
+            // https://developer.android.com/about/versions/oreo/background-location-limits
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                startForegroundService(ForegroundService.getForegroundServiceIntent(Settings.this));
+            } else {
+                SettingsReceivers.Register(getApplicationContext());
```

Comment:
This line keeps the current implementation, but this PR doesn't run the register for recording action. The service's onCreate method is only for over Android O